### ISSUE TITLE
AppImage doesn't start due to symbol lookup error

### ISSUE THREAD

**Issue by SirDK** [rancher-sandbox/rancher-desktop#6122]
### Actual Behavior

Rancher Desktop throws appimage throws error on launch and doesn't launch at all.

### Steps to Reproduce

1. Download [latest](https://download.opensuse.org/repositories/isv:/Rancher:/stable/AppImage/) appimage;
2. Launch it.

### Result

Throws error on launch: `rancher-desktop: symbol lookup error: /lib64/libcairo.so.2: undefined symbol: pixman_image_set_dither`. Doesn't launch.

### Expected Behavior

Rancher Desktop appimage launches as good as rpm for openSUSE.

### Additional Information

Tested on openSUSE Tumbleweed with last updates and Fedora 39 Kinoite. Same error in both situations. I can note that on openSUSE Tumbleweed rpm package works correct. Based on the [description](https://docs.rancherdesktop.io/getting-started/installation/#installing-via-rpm-package:~:text=Note%3A%20RHEL%2C%20Fedora%2C%20and%20related%20distributions%20package%20QEMU%2C%20which%20Rancher%20Desktop%20uses%20on%20Linux%2C%20differently%20than%20other%20distributions.%20To%20use%20Rancher%20Desktop%20on%20these%20distributions%2C%20please%20use%20the%20AppImage.) of the rpm package, they are not compatible with RedHat distributions due to differences in QEMU packaging. I would like to be able to not only run appimage in general, but also on Fedora. Because appimage implies it.

### Rancher Desktop Version

1.11.1-Build4.1.glibc2.27-x86_64

### Rancher Desktop K8s Version

***

### Which container engine are you using?

containerd (nerdctl)

### What operating system are you using?

Other Linux

### Operating System / Build Version

Fedora Linux 39 Kinoite 6.6.3-666.rog.fc39.x86_64

### What CPU architecture are you using?

x64

### Linux only: what package format did you use to install Rancher Desktop?

AppImage

**Comment by adamkpickering** [comment]
Thanks for creating an issue!

Unfortunately, AppImages are not perfect: in order to run the application, we need certain shared objects (often of a specific version) to be present. These are not always present on the system, so AppImages package a bunch of them by default. However, this is more of a stop-gap solution than a real solution. Sometimes the user needs to perform other installation steps, and sometimes there is just no satisfactory solution.

I installed a Fedora Kinoite 39 VM and tried a few things:
1. Installing the AppImage on the base system as it appears you did. I was able to reproduce your error about `libcairo`.
2. Installing the AppImage into a `toolbox` container. When run, it produced an error about not being able to mount using FUSE.
3. Installing the rpm version of Rancher Desktop into a Tumbleweed `distrobox` container. This appears to work.

Here are the steps I followed:
```
rpm-ostree install distrobox
```
Then wait for it to do its thing. Reboot.
```
distrobox create -i opensuse/tumbleweed
distrobox enter tumbleweed
```
Once you're in the `distrobox` container, you can follow the Rancher Desktop installation steps for the rpm
and then launch Rancher Desktop with
```
rancher-desktop
```
I did some minimal testing (just checking that RD starts up and a `nerdctl run hello-world`) and it appears to work.

Do these steps work for you?

As a sidenote: I know this isn't ideal. We have been wanting to take another crack at providing RD as a flatpak application, which would be ideal, but other things have taken priority.

**Comment by SirDK** [comment]
Thanks for the detailed answer. I didn’t immediately think about distrobox, because containerization can cause caveats inside the container, and I turned out to be right. I performed the actions you described. Additional manipulations are required. There are problems that cannot be solved on my part or even those that are worthy of a separate report (although I doubt the functionality inside the container is guaranteed).
I like the flatpak idea. In my opinion this is better than appimage as a universal package.
Perhaps I’ll turn my attention to solutions for fedora. I will be following your progress on the flatpak topic.
Closing bug since problem in shared libraries and is somewhat related to appimage as format.
